# Copyright (c) Facebook, Inc. and its affiliates.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# TODO: Remove the FixedString.h hunk after
# https://github.com/facebook/folly/pull/2726 is merged and Velox's pinned
# Folly version includes the fix. Keep the dependency-discovery hunks below.
--- a/folly/FixedString.h
+++ b/folly/FixedString.h
@@ -2848,6 +2848,6 @@
 #ifndef NO_FIXED_STR_UDL
 #define FOLLY_DEFINE_FIXED_STRING_UDL(N)                     \
-  constexpr FixedString<N> operator""_fs##N(                 \
+  constexpr FixedString<N> operator"" _fs##N(                \
       const char* that, std::size_t count) noexcept(false) { \
     return {that, count};                                    \
   }                                                          \
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -63,8 +63,8 @@
-find_package(Gflags MODULE)
-set(FOLLY_HAVE_LIBGFLAGS ${LIBGFLAGS_FOUND})
-if(LIBGFLAGS_FOUND)
-  list(APPEND FOLLY_LINK_LIBRARIES ${LIBGFLAGS_LIBRARY})
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBGFLAGS_INCLUDE_DIR})
-  set(FOLLY_LIBGFLAGS_LIBRARY ${LIBGFLAGS_LIBRARY})
-  set(FOLLY_LIBGFLAGS_INCLUDE ${LIBGFLAGS_INCLUDE_DIR})
+find_package(gflags)
+set(FOLLY_HAVE_LIBGFLAGS ${gflags_FOUND})
+if(gflags_FOUND)
+  list(APPEND FOLLY_LINK_LIBRARIES ${gflags_LIBRARY})
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${gflags_INCLUDE_DIR})
+  set(FOLLY_LIBGFLAGS_LIBRARY ${gflags_LIBRARY})
+  set(FOLLY_LIBGFLAGS_INCLUDE ${gflags_INCLUDE_DIR})
 endif()
@@ -72,7 +72,7 @@
-find_package(Glog MODULE)
-set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
+find_package(glog)
+set(FOLLY_HAVE_LIBGLOG ${glog_FOUND})
 list(APPEND FOLLY_LINK_LIBRARIES glog::glog)
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${glog_INCLUDE_DIR})
 # Glog 0.7+ requires GLOG_USE_GLOG_EXPORT to be defined so that headers
 # include glog/export.h which defines GLOG_EXPORT.
 if (EXISTS "${GLOG_INCLUDE_DIR}/glog/export.h")
